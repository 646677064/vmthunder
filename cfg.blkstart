dev=$1
outfile=$2

umount /sys/kernel/debug/
mount -t debugfs debugfs /sys/kernel/debug/
echo stacktrace > /sys/kernel/debug/tracing/trace_options

#mkdir tmp
blktrace -d $dev -o - | blkparse -i - > /tmp/$outfile &
