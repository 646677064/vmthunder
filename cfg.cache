cachesize=$1
iscsitname=$2
parentip=$3
cacheloopdev=$4

sleep 0.2s
#blockdev --setra 0 $cacheoridev

cachedev=`iscsiadm -m session -n $iscsitname.$parentip -P 3 | sed -n 's/^\s\+Attached scsi disk \(sd\w\+\)\s.*$/\1/p'`
cacheoridev="/dev/$cachedev"

flashcache_create -p back -s ${cachesize}m -b 4k cachedev $cacheloopdev $cacheoridev
sleep 0.2s
#blockdev --setra 0 /dev/mapper/cachedev
#blockdev --setra 1024 /dev/mapper/cachedev

