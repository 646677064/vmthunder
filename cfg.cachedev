cachesize=$1

mkdir /root/sftsan/ramfs/ 2>/dev/null
mount -t ramfs -o size=${cachesize}m ramfs /root/sftsan/ramfs/
dd if=/dev/zero of=/root/sftsan/ramfs/ramfs.bd bs=1M count=$cachesize
losetup /dev/loop6 /root/sftsan/ramfs/ramfs.bd 
