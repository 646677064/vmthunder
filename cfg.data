imagenode=192.168.0.152

imagename=ubuntu12
numsnap=2
vgname=vgdata
imagedev=/dev/mapper/vgdata-lvdata1

degree=1
workdir=/root/sftsan/

nodes=`cat $workdir/nodes`

cachesize=4096
cacheoridev=/dev/sdh
cacheloopdev=/dev/loop6

dmsnap=dmsnap
qcsnap=qcsnap
snaptype=$dmsnap
#snaptype=$qcsnap
