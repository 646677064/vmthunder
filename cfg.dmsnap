imagename=$1
numsnap=$2
vgname=$3
workdir=$4

modprobe dm_snapshot
for isnap in `seq $numsnap `
do
	snapdev=sdev_${imagename}_$isnap
	snaporigin=ori-${imagename}_$isnap
	snapname=snp-${imagename}_$isnap

	sh $workdir/cfg.lvm $vgname $snapdev
	snapdev=$vgname-$snapdev
        
        echo $snapdev $snaporigin $snapname

	echo 0 `blockdev --getsz /dev/mapper/cachedev` snapshot-origin /dev/mapper/cachedev | dmsetup create $snaporigin
	echo 0 `blockdev --getsz /dev/mapper/cachedev` snapshot /dev/mapper/$snaporigin /dev/mapper/$snapdev N 128 | dmsetup create $snapname
	blockdev --setra 0 /dev/mapper/$snapname

done
