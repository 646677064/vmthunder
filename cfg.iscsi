iscsiadm -m discovery -t sendtargets -p $1 
iscsiadm -m node -T $2 -p $1 --login
