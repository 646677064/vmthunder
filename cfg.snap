imagename=$1
numsnap=$2
workdir=$3

modprobe dm_snapshot
for isnap in `seq $numsnap `
do

	snapdev=sdev_${imagename}_$isnap
	snaporigin=ori-${imagename}_$isnap
	snapname=snp-${imagename}_$isnap

	sh $workdir/cfg.lvm $snapdev
	snapdev=vg_newpdl-$snapdev

	echo 0 `blockdev --getsz /dev/mapper/cachedev` snapshot-origin /dev/mapper/cachedev | dmsetup create $snaporigin
	echo 0 `blockdev --getsz /dev/mapper/cachedev` snapshot /dev/mapper/$snaporigin /dev/mapper/$snapdev N 128 | dmsetup create $snapname

done
